azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java

```diff
@@ -509,6 +523,66 @@ public void testPodConstruction() throws Exception {
         .getVolumeMounts().size());
   }
 
+  @Test
+  public void testPopulatingProxyUsersFromProject() throws Exception {
```

Comment:
[non-blocker] we are testing three scenarios using this single test. Having three separate tests would make it easy to reason about. Did we consider that @utk-12?

Reply:
Ack. Will use 3 separate tests. thank you !